File: lesson2/DemoWebApplication/Components/WeatherForecastComponent/Services/OpenWeatherForecastService.cs
using DemoWebApplication.Components.WeatherForecastComponent.Dtos;
using DemoWebApplication.Components.WeatherForecastComponent.Models;
using DemoWebApplication.Components.WeatherForecastComponent.Services.Interfaces;
using Newtonsoft.Json;

namespace DemoWebApplication.Components.WeatherForecastComponent.Services;

public class OpenWeatherForecastService : IWeatherForecastService
{
    public WeatherForecastDto GetByLocation(string location)
    {
        var apiUrl =
            "https://api.openweathermap.org/data/2.5/weather?lat=44.34&lon=10.99&appid=5796abbde9106b7da4febfae8c44c232";
        
        using var httpClient = new HttpClient();
        var resp = httpClient.GetAsync(apiUrl).Result;
        var json = resp.Content.ReadAsStringAsync().Result;

        var model = JsonConvert.DeserializeObject<OpenWeatherMapModel>(json);

        return new WeatherForecastDto()
        {
            Location = model.Name,
            Humidity = model.Main.Humidity,
            Pressure = model.Main.Pressure,
            TempMax = model.Main.TempMax,
            TempMin = model.Main.TempMin
        };
    }

    public async Task<WeatherForecastDto> GetByLocationAsync(string location)
    {
        var apiUrl =
            "https://api.openweathermap.org/data/2.5/weather?lat=44.34&lon=10.99&appid=5796abbde9106b7da4febfae8c44c232";

        using var httpClient = new HttpClient();
        var resp = await httpClient.GetAsync(apiUrl);
        var json = await resp.Content.ReadAsStringAsync();

        var model = JsonConvert.DeserializeObject<OpenWeatherMapModel>(json);

        return new WeatherForecastDto()
        {
            Location = model.Name,
            Humidity = model.Main.Humidity,
            Pressure = model.Main.Pressure,
            TempMax = model.Main.TempMax,
            TempMin = model.Main.TempMin
        };
    }
}

File: lesson2/DemoNestJsApplication/src/components/weather-forecast-component/services/open-weather-forecast.service.ts

import axios from 'axios';
import { IWeatherForecastService } from './interfaces/i.weather-forecast.service';
import { WeatherForecastDto } from '../dtos/weather-forecast.dto';
import { error } from 'console';

interface OpenWeatherMapModel {
    name: string;
    main: {
        humidity: number;
        pressure: number;
        temp_max: number;
        temp_min: number;
    };
}

export class OpenWeatherForecastService implements IWeatherForecastService {
    private apiUrl: string = "https://api.openweathermap.org/data/2.5/weather?lat=44.34&lon=10.99&appid=5796abbde9106b7da4febfae8c44c232";

    getByLocation(location: string): WeatherForecastDto {
        throw 'Not initialized.';
    }

    async getByLocationAsync(location: string): Promise<WeatherForecastDto> {
        const response = await axios.get(this.apiUrl);
        const model = response.data;
  
        return {
          location: model.name,
          humidity: model.main.humidity,
          pressure: model.main.pressure,
          tempMax: model.main.temp_max,
          tempMin: model.main.temp_min,
        };
    }
}
